modin/engines/base/io/text/csv_reader.py

```diff
@@ -126,8 +129,10 @@ def _read(cls, filepath_or_buffer, **kwargs):
             # This is the number of splits for the columns
             num_splits = min(len(column_names), num_partitions)
             # This is the chunksize each partition will read
-            chunk_size = max(1, (total_bytes - f.tell()) // num_partitions)
-
+            if nrows is None:
+                part_size = max(1, total_bytes // num_partitions)
+            else:
+                part_size = max(1, (nrows - skiprows) // num_partitions)
```

Comment:
It's strange that in one case `part_size` is computed by bytes, in another by line count.

Reply:
changed it in the latest commit, now we have `chunk_size_bytes` and `rows_per_part` variable, and depends on `nrows` one of those variables is `None` because it could not be computed (actually we can compute it, but that will give us unnecessary overhead: about ~2s in case of 2gb file). 

We pass both of these variables to `read_rows` which decides which one to use